cm_ovirt_vm_creator.py

```diff
@@ -143,46 +143,83 @@ def construct_search_by_name_query(node_name):
     return search_string.format(node_name=node_name).__str__()
 
 
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in range(0, len(l), n):
+        yield l[i:i + n]
+
+
+def start_vm(vm_service, args):
+    """ Start a VM """
+    pub_sshkey = os.environ[args.pub_sshkey]
+    # ovirt API quirk - starting a VM with wait=False does nothing,
+    # and unlike other APIs, it accept async=True to make it asynchronous
```

Comment:
`wait=False` tells the *client* to not wait for the response. `async=True` tells the *server* to not wait for the response.

The `async` parameter isn't reliable, try to avoid it.

When you use `wait=True` nothing happens till you do *any* `future.wait()` call. When you do that call all the pending requests will be sent (up to a configurable limit), and responses processed, till the response for the one that you are waiting for is received.

So I think you should change this to use only `wait=False`, and then wait for the corresponding futures.